lib/glific/contacts.ex

```diff
@@ -275,9 +275,22 @@ defmodule Glific.Contacts do
   end
 
   @doc """
-  Check if this contact id is a new conatct
+  This function is called by the messaging framework for all incoming messages, hence
+  might be a good candidate to maintain a contact level cache at some point
+
+  We use a fetch followed by create, to avoid the explosion in the id namespace
   """
+  @spec maybe_create_contact(map()) :: {:ok, Contact.t()} | {:error, Ecto.Changeset.t()}
+  def maybe_create_contact(sender) do
+    case Repo.get_by(Contact, %{phone: sender.phone}) do
+      nil -> create_contact(sender)
+      contact -> {:ok, contact}
```

Comment:
I think It will not update the contact name if that is changed on there whats app.

Reply:
yes, i realized that, and seems might be worth it, since folks don't change their whatsapp name often, maybe we can put a check to see if the name is different and if so, do an update. yes, this sounds better

Reply:
Sure.

Reply:
i fixed this, since its a conditional update